model/device_deployment.go

```diff
@@ -23,44 +23,153 @@ import (
 	"github.com/pkg/errors"
 )
 
-type DeviceDeploymentStatus string
+// DeviceDeploymentStatus is an enumerated type showing the status of a device within a deployment
+type DeviceDeploymentStatus int32
 
 // Deployment statuses
 const (
-	DeviceDeploymentStatusDownloading        DeviceDeploymentStatus = "downloading"
-	DeviceDeploymentStatusInstalling         DeviceDeploymentStatus = "installing"
-	DeviceDeploymentStatusRebooting          DeviceDeploymentStatus = "rebooting"
-	DeviceDeploymentStatusPending            DeviceDeploymentStatus = "pending"
-	DeviceDeploymentStatusSuccess            DeviceDeploymentStatus = "success"
-	DeviceDeploymentStatusFailure            DeviceDeploymentStatus = "failure"
-	DeviceDeploymentStatusNoArtifact         DeviceDeploymentStatus = "noartifact"
-	DeviceDeploymentStatusAlreadyInst        DeviceDeploymentStatus = "already-installed"
-	DeviceDeploymentStatusAborted            DeviceDeploymentStatus = "aborted"
-	DeviceDeploymentStatusDecommissioned     DeviceDeploymentStatus = "decommissioned"
-	DeviceDeploymentStatusPauseBeforeInstall DeviceDeploymentStatus = "pause_before_installing"
-	DeviceDeploymentStatusPauseBeforeCommit  DeviceDeploymentStatus = "pause_before_committing"
-	DeviceDeploymentStatusPauseBeforeReboot  DeviceDeploymentStatus = "pause_before_rebooting"
+	// The following statuses are distributed evenly by incrementing the
+	// enum counter's second byte.
+	// NOTE: when adding new statuses into the list, use the mean value between the
+	//       neighbouring values and append the value AFTER the following list and extend
+	//       the DeviceDeploymentStatus<type>Str constant and allStatuses variable as well
+	//       as the MarshalText and UnmarshalText interface functions.
+	//       See example below.
+	// WARN: DO NOT CHANGE ANY OF THE FOLLOWING VALUES.
+	DeviceDeploymentStatusNull DeviceDeploymentStatus = iota << 8 // i=0... {i * 2^8}
+	DeviceDeploymentStatusFailure
+	DeviceDeploymentStatusAborted
+	DeviceDeploymentStatusPauseBeforeInstall
+	DeviceDeploymentStatusPauseBeforeCommit
+	DeviceDeploymentStatusPauseBeforeReboot
+	DeviceDeploymentStatusDownloading
+	DeviceDeploymentStatusInstalling
+	DeviceDeploymentStatusRebooting
+	DeviceDeploymentStatusPending
+	DeviceDeploymentStatusSuccess
+	DeviceDeploymentStatusNoArtifact
+	DeviceDeploymentStatusAlreadyInst
+	DeviceDeploymentStatusDecommissioned
+	// DeviceDeploymentStatusNew = (DeviceDeploymentStatusSuccess + DeviceDeploymentStatusNoArtifact) / 2
```

Comment:
Will this be easy to maintain in the future if we start adding new statuses?
Probably not an issue today, just wondering if we shouldn't make it easier:

DeviceDeploymentStatusNull = 1000
DeviceDeploymentStatusFailure = 2000
DeviceDeploymentStatusAborted = 3000
...

Reply:
Right now they are:
```go
DeviceDeploymentStatusNull = 0
DeviceDeploymentStatusFailure = 512
DeviceDeploymentStatusAborted = 1024
DeviceDeploymentStatusPauseBeforeInstall = 1536
// next = prev + 512
```
Maybe the `iota` expression makes it unclear, I just really felt like using it because I rarely ever do :sweat_smile: